layers/+window-management/exwm/funcs.el

```diff
@@ -78,6 +78,68 @@ Can show completions at point for COMMAND using helm"
   (interactive)
   (start-process "" nil exwm-locking-command))
 
+
+(defvar exwm//autostart-process-list nil
+  "List of processes run during autostart.")
+
+(defun exwm/autostart-process (name command &optional directory)
+  "Can be used during initialization to run COMMAND as a process
+  with NAME and add it to the list of autostarted processes.
+
+DIRECTORY can be set to a string which will be used as working directory for the
+process.  If not supplied, will be set to `user-home-directory'.
+"
+  (push (let ((default-directory (or directory user-home-directory)))
+          (start-process-shell-command name nil command))
+        exwm//autostart-process-list))
+
+(defun exwm//start-desktop-application (basename xdg)
+  "Autostart an application from a XDG desktop entry specification."
+  (let* ((type (gethash "Type" xdg))
+         ;; (name (gethash "Name" xdg))
+         (cmd (gethash "Exec" xdg))
+         (hidden (gethash "Hidden" xdg))
+         (include (gethash "OnlyShowIn" xdg))
+         (exclude (gethash "NotShowIn" xdg))
+         (try-exec (gethash "TryExec" xdg))
+         (exec-directory (gethash "Path" xdg))
+         ;; (dbus-p (gethash "DBusActivatable" xdg)) ; TODO: support
+         (included-p (cond (include (member "EXWM" (split-string include ";" t)))
+                           (exclude (not (member "EXWM" (split-string exclude ";" t))))
+                           (t)))
+         (should-exec-p (and
```

Comment:
This is very inefficient.
You should check these conditions much earlier.
Consider change `let*` to `when-let*`. For example:

`(when-let ((type (gethash "Type" xdg)) (_ (string= type "Appliction")))`

If `type` is nil, it ends early.
If `type` doesn't equal "Application", it ends early.